packages/ketcher-react/src/script/ui/state/moveSelectedItems.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  Editor,
+  Vec2,
+  EditorSelection,
+  fromMultipleMove,
+  scrollByVector,
+  ReAtom
+} from 'ketcher-core'
+
+type ArrowKey = 'ArrowUp' | 'ArrowDown' | 'ArrowRight' | 'ArrowLeft'
+const distinationVectorMapping: { [key in ArrowKey]: Vec2 } = {
+  ArrowUp: new Vec2(0, -1),
+  ArrowDown: new Vec2(0, 1),
+  ArrowRight: new Vec2(1, 0),
+  ArrowLeft: new Vec2(-1, 0)
+}
+
+export function isArrowKey(key: string): key is ArrowKey {
+  return (
+    key === 'ArrowUp' ||
+    key === 'ArrowDown' ||
+    key === 'ArrowLeft' ||
+    key === 'ArrowRight'
+  )
+}
+
+export function moveSelectedItems(
+  editor: Editor,
+  key: ArrowKey,
+  isShiftPressed: boolean
+) {
+  const stepFactor = 1 / editor.options().scale
+  const fasterStepFactor = stepFactor * 10
+  const selectedItems = editor.explicitSelected()
+  const distinationVector = distinationVectorMapping[key]
+  const action = fromMultipleMove(
+    editor.render.ctab,
+    selectedItems,
+    distinationVector.scaled(isShiftPressed ? fasterStepFactor : stepFactor)
+  )
+  editor.update(action, false, { resizeCanvas: true })
+  const isClose = isCloseToTheEdgeOfCanvas(selectedItems, editor, key)
+  if (isClose) {
+    scrollByVector(distinationVector, editor.render)
+  }
+  editor.rotateController.rerender()
```

Comment:
Just curious why we need to use the rotate controller here?

Reply:
Because when we select structure "rotateController" appears. And then, we press some arrow button, structure is moved and rotateController doesn't move with the selected items